ci-operator/step-registry/cucushift/upgrade/arch-migration/cucushift-upgrade-arch-migration-commands.sh

```diff
@@ -359,11 +396,25 @@ fi
 export OC="run_command_oc"
 
 SOURCE_VERSION="$(oc get clusterversion --no-headers | awk '{print $2}')"
-SOURCE_XY_VERSION="$(echo "${SOURCE_VERSION}" | cut -f1,2 -d.)"
 export SOURCE_VERSION
+SOURCE_MINOR_VERSION="$(echo "${SOURCE_VERSION}" | cut -f2 -d.)"
+export SOURCE_MINOR_VERSION
+SOURCE_XY_VERSION="$(echo "${SOURCE_VERSION}" | cut -f1,2 -d.)"
+export SOURCE_XY_VERSION
+if [[ "${SOURCE_MINOR_VERSION}" -gt 17 ]]; then
+    OLD_OPERATOR_IMAGE="$(oc get co/machine-config -oyaml | yq -r '.status.versions[] | select(.name=="operator-image") | .version')"
+    export OLD_OPERATOR_IMAGE
+fi
 
 switch_channel
 migrate
+if [[ "${SOURCE_MINOR_VERSION}" -gt 17 ]]; then
+    wait_upgrade_start
+fi
 check_migrate_status
 check_arch
-health_check
+if [[ "${SOURCE_MINOR_VERSION}" -gt 17 ]]; then
+    check_co
+else
+    health_check
```

Comment:
We can not do health check any more when version > 17 ?

Reply:
to make sure the cluster is ready for following e2e, we need to do health check, the point of this fix pr is to make sure the cluster is able to add additional arch nodes immediately after the clusterversion shows upgrade is completed.

Reply:
Look at the test plan on https://issues.redhat.com/browse/OTA-1386 and https://github.com/openshift/enhancements/blob/master/enhancements/update/duration-of-migration-to-multi-arch.md#test-plan, this is required for 4.17+

Reply:
That's because our customer will not check if mcp ready, if co ready, they add extra nodes once they see cluster upgrade is finished.

Reply:
> our customer will not check if mcp ready, if co ready,

When mcp is being updating, machine-config co would be shown as "ready" or "available, but updating" ?

Reply:
If I remember correct, there was Jenkins CI bug that mcp not ready but "ClusterVersion claims completeness"

Reply:
Maybe we can always dump out `oc get machineconfigpools` and `oc get co` in `debug` function, in order to easily observe the behavior.

Reply:
Sure

Reply:
I checked the debug method, it already have that output:
```
        echo -e "Describing abnormal operators...\n"
        oc get co --no-headers | awk '$3 != "True" || $4 != "False" || $5 != "False" {print $1}' | while read co; do echo -e "\n#####oc describe co ${co}#####\n$(oc describe co ${co})"; done
        echo -e "Describing abnormal mcp...\n"
        oc get machineconfigpools --no-headers | awk '$3 != "True" || $4 != "False" || $5 != "False" {print $1}' | while read mcp; do echo -e "\n#####oc describe mcp ${mcp}#####\n$(oc describe mcp ${mcp})"; done
```

Reply:
There is a condition - `(( FRC != 0 ))`, we can add `oc get co` and `oc get machineconfigpools` outside the if block

Reply:
ok